packages/teams-js/test/public/pages.spec.ts

```diff
@@ -210,42 +335,917 @@ describe('AppSDK-TeamsAPIs', () => {
+          expect(promise).resolves.toBe(expectedTabInstanceParameters);
+        });
+
+        it(`pages.tabs.getTabInstances should be undefined when no parameters are passed and initialized with ${context} context`, async () => {
+          await utils.initializeWithContext(context);
+          const promise = pages.tabs.getTabInstances();
+          let message = utils.findMessageByFunc('getTabInstances');
+
+          utils.respondToMessage(message);
+          expect(message).not.toBeNull();
+          expect(promise).resolves.toBeUndefined();
+        });
+      });
+    });
+
+    describe('Testing pages.tabs.getMruTabInstances function', () => {
+      let expectedTabInstanceParameters: TabInstanceParameters = {
+        favoriteChannelsOnly: true,
+        favoriteTeamsOnly: true,
+      };
+
+      it('pages.tabs.getMruTabInstances should not allow calls before initialization', async () => {
+        await expect(() => pages.tabs.getMruTabInstances()).rejects.toThrowError(
+          'The library has not yet been initialized',
+        );
+      });
+
+      Object.values(FrameContexts).forEach(context => {
+        it(`pages.tabs.getMruTabInstances should successfully getTabInstance when no parameters are passed and initialized with ${context} context`, async () => {
+          await utils.initializeWithContext(context);
+          const promise = pages.tabs.getMruTabInstances();
+          let message = utils.findMessageByFunc('getMruTabInstances');
+
+          utils.respondToMessage(message, expectedTabInstanceParameters);
+          expect(message).not.toBeNull();
+          expect(promise).resolves.toBe(expectedTabInstanceParameters);
+        });
+
+        it(`pages.tabs.getMruTabInstances should be undefined getTabInstance when parameters are passed and  initialized with ${context} context`, async () => {
+          await utils.initializeWithContext(context);
+          const promise = pages.tabs.getMruTabInstances(expectedTabInstanceParameters);
+          let message = utils.findMessageByFunc('getMruTabInstances');
+
+          utils.respondToMessage(message, expectedTabInstanceParameters);
+          expect(message).not.toBeNull();
+          expect(promise).resolves.toBe(expectedTabInstanceParameters);
+        });
+
+        it(`pages.tabs.getMruTabInstances should be undefined when no parameters are passed and initialized with ${context} context`, async () => {
+          await utils.initializeWithContext(context);
+          const promise = pages.tabs.getMruTabInstances();
+          let message = utils.findMessageByFunc('getMruTabInstances');
+
+          utils.respondToMessage(message);
+          expect(message).not.toBeNull();
+          expect(promise).resolves.toBeUndefined();
+        });
+      });
+    });
+
+    describe('Testing pages.tabs.isSupported function', () => {
+      it('pages.tabs.isSupported should return false if the runtime says pages.tabs is not supported', () => {
```

Comment:
Can you add an additional test that checks that `pages.tabs.isSupported` returns false when `supports: {}` please?

Reply:
Added in the latest commit